server/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -311,12 +310,12 @@ private static Function rewriteAndValidateFields(Function function, Context cont
 
         @Override
         public Query visitReference(Reference ref, Context context) {
+            DataType<?> type = ref.valueType();
             // called for queries like: where boolColumn
-            if (ref.valueType() == DataTypes.BOOLEAN) {
-                if (ref.indexType() == IndexType.NONE) {
-                    return new MatchNoDocsQuery("column does not exist in this index");
-                }
-                return new TermQuery(new Term(ref.storageIdent(), new BytesRef("T")));
+            if (type == DataTypes.BOOLEAN) {
+                // noinspection unchecked, ConstantConditions
+                return ((StorageSupport<Boolean>) type.storageSupport()).eqQuery().termQuery(
+                    ref.storageIdent(), Boolean.TRUE, ref.hasDocValues(), ref.indexType() != IndexType.NONE);
```

Comment:
```suggestion
            if (type == DataTypes.BOOLEAN) {
                StorageSupport<? super Boolean> storageSupport = DataTypes.BOOLEAN.storageSupportSafe();
                return storageSupport.eqQuery().termQuery(
                    ref.storageIdent(),
                    Boolean.TRUE,
                    ref.hasDocValues(),
                    ref.indexType() != IndexType.NONE
                );
```

Avoids all the warnings

Reply:
Thank you, just added another null check for `storageSupport.eqQuery()`.